### PR TITLE
Remove omitempty from mandatory fields

### LINKED ISSUE #426: Doc: Improve the performance profile documentation specifying the Mandatory attributes under PerformanceProfiles
The current documentation of  fields under the PerformanceProfile spec are marked as false under the "Required" .  The Documentation should mark what fields/subset of those fields are required/mandatory/. 

For example in the current performance profile [documentation](https://github.com/openshift-kni/performance-addon-operators/blob/master/docs/performance_profile.md) the fields under CPU are marked as "false" . 

Another example in the case of HugePage which has 2 fields defaultHugepagesSize, pages , it's not clear from the documentation  that if defaultHugepagesSize is specified then pages should also be mandatory field ?. 

Another example is RealTimeKernel where if we mark field "enabled" as true , then the fields under CPU are mandatory, this is not clear from the documentation. 

It would be good if we can mark what fields are mandatory/required  in the documentation.

### PR DESCRIPTION
**Pull request by cynepco3hahue**
Our doc generator set all fields with `omitempty` directive as not
required one. So part of mandatory fields appear as not required under the documentation.

Fixes #426 

Signed-off-by: Artyom Lukianov <alukiano@redhat.com>

### CODE REVIEW

File: api/v2/performanceprofile_types.go [hunk 1]
```diff
@@ -73,8 +73,7 @@ type CPU struct {
 	// except the reserved CPUs. In order to guarantee that your workload will run on the isolated CPU:
 	//   1. The union of reserved CPUs and isolated CPUs should include all online CPUs
 	//   2. The isolated CPUs field should be the complementary to reserved CPUs field
-	// +optional
-	Isolated *CPUSet `json:"isolated,omitempty"`
+	Isolated *CPUSet `json:"isolated"`
```
[the remark this answers is not in the document]
Comment (marcel-apf) [reply api/v2/performanceprofile_types.go:76]: /hold
Just until  @cynepco3hahue answers

File: api/v2/performanceprofile_types.go [hunk 2]
```diff
@@ -73,8 +73,7 @@ type CPU struct {
 	// except the reserved CPUs. In order to guarantee that your workload will run on the isolated CPU:
 	//   1. The union of reserved CPUs and isolated CPUs should include all online CPUs
 	//   2. The isolated CPUs field should be the complementary to reserved CPUs field
```
Comment (fromanirh) [inline api/v2/performanceprofile_types.go:75]: this text is slightly misleasing and we need to improve it:
1. is not true: we had cases on which isolated and reserved were non-overlapping subsets of the online CPUs, and this is supported and should work
2. this is hard to parse, probably simpler to say that the two set MUST not overlap (and nowadays PAO validates that)
I'll file a PR to fix the comment.

**Comment by fromanirh** [comment]
Misread a statement and posted a wrong comment. Removed.

File: api/v2/performanceprofile_types.go [hunk 3]
```diff
@@ -73,8 +73,7 @@ type CPU struct {
 	// except the reserved CPUs. In order to guarantee that your workload will run on the isolated CPU:
```
Comment (fromanirh) [inline api/v2/performanceprofile_types.go:73]: I misread this statement and though it was misleading. It actually isn't, the key here is "in order to guarantee". Then the suggestions are valid because a system configured like that indeed should guarantee that the workload runs on isolated CPUs. Removed my previous comment which added suggestions based on my wrong assumption.

Comment (fromanirh) [reply api/v2/performanceprofile_types.go:73]: let's make sure all the cases are covered: https://github.com/openshift-kni/performance-addon-operators/pull/434

File: api/v2/performanceprofile_types.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (cynepco3hahue) [reply api/v2/performanceprofile_types.go:76]: So currently I removed omitempty only from fields that we strictly request to be presented, otherwise, the validation will fail.
See https://github.com/openshift-kni/performance-addon-operators/blob/af31700a496376c6cf70508ba0e65289479bc127/pkg/controller/performanceprofile/components/profile/profile.go#L23.
Probably we can change it, but the change should be done under the different PR.

[the remark this answers is not in the document]
Comment (marcel-apf) [reply api/v2/performanceprofile_types.go:76]: ok
